OSDOCS-13576: Added info on rosa create network command (#91784)
openshift/openshift-docs: modules/rosa-hcp-create-network.adoc

// Module included in the following assemblies:
//
// * rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc

ifeval::["{context}" == "rosa-hcp-egress-lockdown-install"]
:egress-lockdown-rosa:
endif::[]

:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-create-network_{context}"]
= Creating a Virtual Private Cloud using the ROSA CLI

> The `rosa create network` command is available in v.1.2.48 or later of the ROSA CLI. This command uses AWS CloudFormation to create a VPC and other networking components used to install a ROSA cluster. CloudFormation is a native AWS infrastructure-as-code tool, compatible with the AWS CLI. For more information on AWS CloudFormation, see the additional resources.

Reviewer: ```suggestion
The `rosa create network` command is available in v.1.2.48 or later of the ROSA command-line interface (CLI). The command uses AWS CloudFormation to create a VPC and the other networking components used to install a ROSA cluster. CloudFormation is a native AWS infrastructure-as-code tool and is compatible with the AWS CLI. For more information on AWS CloudFormation, see the additional resources.
```

If you do not specify a template, CloudFormation uses a default template that creates the following parameters:

[cols="2a,3a",options="header"]
|===
|VPC parameter
|Value

| Availability Zones
| 1

| Region 
| `us-east-1`
  
| VPC CIDR 
| `10.0.0.0/16` 
|===

You can create and customize CloudFormation templates can also be used with the command. See the additional resources for information on the Default VPC template.

.Prerequisites

* You have configured your AWS account
* You have configured your Red Hat accounts

> * You have installed and configured the ROSA CLI to the latest version

Reviewer: ```suggestion
* You have installed the ROSA CLI and configured it to the latest version
```

.Procedure

> . Use the following command to create an AWS VPC using the default CloudFormations template:

Reviewer: ```suggestion
. Create an AWS VPC using the default CloudFormations template by running the following command:
```

+
[source,terminal]
----
$ rosa create network
----

> . Optional: If you want to modify a parameter, you can specify `--param Param1=Value1`, `--param Param2=Value2`, and so on. For example, if you wanted to modify the region, name, AZ count, and VPC's CIDR parameters, you could run:

Reviewer: ```suggestion
. Optional: If you want to modify a parameter, you can specify the parameter using optional argument flags such as `--param Param1=Value1` and `--param Param2=Value2`. For example, if you want to modify the region, name, availability zone (AZ) count, and VPC's CIDR parameters, you could run:
```

+
[source,terminal]
----
$ rosa create network --param Region=us-west-2 --param Name=quickstart-stack --param AvailabilityZoneCount=1 --param VpcCidr=10.0.0.0/16
----

> . The full list of parameters available in the default template.

Reviewer: ```suggestion
. The full list of parameters is available in the default template.
```

+
.Example template
+
[source,yaml]
----
Parameters:
  AvailabilityZoneCount:
    Type: Number
    Description: "Number of Availability Zones to use"
    Default: 1
    MinValue: 1
    MaxValue: 3
  Region:
    Type: String
    Description: "AWS Region"
    Default: "us-west-2"
  Name:
    Type: String
    Description: "Name prefix for resources"
  VpcCidr:
    Type: String
    Description: CIDR block for the VPC
    Default: '10.0.0.0/16'
----
+

> The command takes about 5 minutes to run, providing regular status updates from AWS as resources are created. If there is an issue with CloudFormation, a rollback will be attempted. If any other issues are encountered, please follow the error message instructions or contact AWS support.

Reviewer: ```suggestion
The command takes about 5 minutes to run and provides regular status updates from AWS as resources are created. If there is an issue with CloudFormation, a rollback is attempted. For all other errors, follow the error message instructions or contact AWS support.
```

+
[NOTE]
====

> Save the subnet IDs that are created as a result of this command since these subnets are needed when creating the cluster. Replace `$SUBNET_IDS` in the `rosa cluster create` command below with these values in a comma-separated list.

Reviewer: ```suggestion
Save the subnet IDs that are created as a result of this command since these subnet IDs are needed to create the cluster. Replace `$SUBNET_IDS` in the `rosa cluster create` command below with these values in a comma-separated list.
```

> During the cluster creation, you can select your create network stack with the following step:

Reviewer: ```suggestion
During cluster creation, you can select your create network stack by using the following step:
```

[source,terminal]
----
? Additional 'Compute' Security Group IDs (optional): sg-0d1fba92a71d6cc6d ('rosa-network-stack-269733383066-SecurityGroup-zD8qkxnh6wwL')
----
====

> . To delete the Cloudformation stack that was created, run the following command:

Reviewer: ```suggestion
. Delete the CloudFormation stack by running the following command:
```

[id="rosa-hcp-create-network_{context}"]
= Creating a Virtual Private Cloud using the ROSA CLI

The `rosa create network` command is available in v.1.2.48 or later of the ROSA command-line interface (CLI). The command uses AWS CloudFormation to create a VPC and the other networking components used to install a ROSA cluster. CloudFormation is a native AWS infrastructure-as-code tool and is compatible with the AWS CLI.

If you do not specify a template, CloudFormation uses a default template that creates the following parameters:

[cols="2a,3a",options="header"]
|===
|VPC parameter
|Value

| Availability zones
| 1

| Region 
| `us-east-1`
  
| VPC CIDR 
| `10.0.0.0/16` 
|===

You can create and customize CloudFormation templates to use with the `rosa create network` command. See the additional resources for information on the default VPC template.

.Prerequisites

* You have configured your AWS account
* You have configured your Red Hat accounts
* You have installed the ROSA CLI and configured it to the latest version

.Procedure
. Create an AWS VPC using the default CloudFormations template by running the following command:
+
[source,terminal]
----
$ rosa create network
----

> . Optional: If you want to modify a parameter, you can specify the parameter using optional argument flags such as `--param Param1=Value1` and `--param Param2=Value2`. For example, if you wanted to modify the region, name, availability zone (AZ) count, and VPC's CIDR parameters, you could run:

Reviewer 2: I'd suggest modifying this step so that it is phrased as a clear step followed by an explanation, e.g.

```
. Optional: Customize your VPC by specifying additional parameters.
+
You can use the `--param` flag to specify changes to the default VPC template. The following example command specifies custom values for `region`, `Name`, `AvailabilityZoneCount` and `VpcCidr`.
```

+
[source,terminal]
----

> $ rosa create network --param Region=us-west-2 --param Name=quickstart-stack --param AvailabilityZoneCount=1 --param VpcCidr=10.0.0.0/16

Reviewer 2: Shall we  modify the AvailabilityZoneCount here to 2 or 3 so they have a working example?

----

> . The full list of parameters is available in the default template.

Reviewer 2: This should probably not be in the procedure. It might be more appropriate in the `rosa create network` CLI reference? Or at least in a reference module.

// Module included in the following assemblies:
//
// * rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc

ifeval::["{context}" == "rosa-hcp-egress-lockdown-install"]
:egress-lockdown-rosa:
endif::[]

:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-create-network_{context}"]
= Creating a Virtual Private Cloud using the ROSA CLI

> The `rosa create network` command is available in v.1.2.48 or later of the ROSA command-line interface (CLI). The command uses AWS CloudFormation to create a VPC and the other networking components used to install a ROSA cluster. CloudFormation is a native AWS infrastructure-as-code tool and is compatible with the AWS CLI. For more information on AWS CloudFormation, see the additional resources.

Reviewer 2: I believe we generally avoid referring folks to the additional resources section - might be worth omitting the final sentence here.

INFO[0140] Resource: ElasticIP1, Type: AWS::EC2::EIP, ID: <IP>
INFO[0140] Resource: ElasticIP2, Type: AWS::EC2::EIP, ID: <IP> 
INFO[0140] Resource: InternetGateway, Type: AWS::EC2::InternetGateway, ID: igw-016e1a71b9812464e 
INFO[0140] Resource: KMSVPCEndpoint, Type: AWS::EC2::VPCEndpoint, ID: <vpce-id> 
INFO[0140] Resource: NATGateway1, Type: AWS::EC2::NatGateway, ID: <nat-gateway-id> 
INFO[0140] Resource: PrivateRoute, Type: AWS::EC2::Route, ID: <route-id> 
INFO[0140] Resource: PrivateRouteTable, Type: AWS::EC2::RouteTable, ID: <route-id> 
INFO[0140] Resource: PrivateSubnetRouteTableAssociation1, Type: AWS::EC2::SubnetRouteTableAssociation, ID: <route-id>
INFO[0140] Resource: PublicRoute, Type: AWS::EC2::Route, ID: <route-id> 
INFO[0140] Resource: PublicRouteTable, Type: AWS::EC2::RouteTable, ID: <route-id> 
INFO[0140] Resource: PublicSubnetRouteTableAssociation1, Type: AWS::EC2::SubnetRouteTableAssociation, ID: <route-id> 
INFO[0140] Resource: S3VPCEndpoint, Type: AWS::EC2::VPCEndpoint, ID: <vpce-id> 
INFO[0140] Resource: STSVPCEndpoint, Type: AWS::EC2::VPCEndpoint, ID: <vpce-id> 
INFO[0140] Resource: SecurityGroup, Type: AWS::EC2::SecurityGroup, ID: <security-group-id> 

> INFO[0140] Resource: SubnetPrivate1, Type: AWS::EC2::Subnet, ID: <subnet-id-1> \ <1>

Reviewer 2: Probably worth using `<private-subnet-id-1>` and `<public-subnet-id-1>` here to distinguish between them

INFO[0140] Resource: SubnetPublic1, Type: AWS::EC2::Subnet, ID: <subnet-id-1> \ <1>
INFO[0140] Resource: VPC, Type: AWS::EC2::VPC, ID: <vpc-id>
INFO[0140] Stack rosa-network-stack-5555 created \ <2>
----
+
--
<1> These two subnet IDs are used to create your cluster when using the `rosa create cluster` command.
<2> The network stack name is used to delete the resource later.
+

> [NOTE]

Reviewer 2: This note should probably be a part of interactive CLI cluster creation rather than here? Also, why would you want to do this? When might it be necessary?

====
Optional: You can add the created network stack to your compute security IDs by selecting the network stack when prompted as follows:
[source,terminal]
----
? Additional 'Compute' Security Group IDs (optional):  [Use arrows to move, space to select, <right> to all, <left> to none, type to filter, ? for more help]
> [ ]  sg-<security-group-id> ('rosa-network-stack-5555-SecurityGroup-aaa')
----
====
--

> . Delete the CloudFormation stack by running the following command:

Reviewer 2: Is this part of creating the network, or is this how you would get rid of a network you had previously created? Should it move to a separate module?

[cols="2a,3a",options="header"]
|===
|VPC parameter
|Value

| Availability zones
| 1

| Region 
| `us-east-1`
  
| VPC CIDR 
| `10.0.0.0/16` 
|===

You can create and customize CloudFormation templates to use with the `rosa create network` command. See the additional resources of this section for information on the default VPC template.

.Prerequisites

* You have configured your AWS account
* You have configured your Red Hat accounts
* You have installed the ROSA CLI and configured it to the latest version

.Procedure
. Create an AWS VPC using the default CloudFormations template by running the following command:
+
[source,terminal]
----
$ rosa create network
----

. Optional: Customize your VPC by specifying additional parameters.
+
You can use the `--param` flag to specify changes to the default VPC template. The following example command specifies custom values for `region`, `Name`, `AvailabilityZoneCount` and `VpcCidr`.
+
[source,terminal]
----
$ rosa create network --param Region=us-west-2 --param Name=quickstart-stack --param AvailabilityZoneCount=3 --param VpcCidr=10.0.0.0/16
----
+

> The command takes about 5 minutes to run and provides regular status updates from AWS as resources are created. If there is an issue with CloudFormation, a rollback is attempted. For all other errors are encountered, please follow the error message instructions or contact AWS support.

Reviewer 3: ```suggestion
The command takes about 5 minutes to run and provides regular status updates from AWS as resources are created. If there is an issue with CloudFormation, a rollback is attempted. For all other errors that are encountered, please follow the error message instructions or contact AWS support.
```
